hathor/transaction/resources/block_at_height.py

```diff
@@ -0,0 +1,173 @@
+
+import json
+from typing import TYPE_CHECKING
+
+from twisted.web import resource
+
+from hathor.api_util import get_missing_params_msg, parse_get_arguments, set_cors
+from hathor.cli.openapi_files.register import register_resource
+
+if TYPE_CHECKING:
+    from twisted.web.http import Request
+
+    from hathor.manager import HathorManager
+
+
+@register_resource
+class BlockAtHeightResource(resource.Resource):
+    """ Implements a web server API to return the block at specific height.
+
+    You must run with option `--status <PORT>`.
+    """
+    isLeaf = True
+
+    def __init__(self, manager: 'HathorManager'):
+        # Important to have the manager so we can know the tx_storage
+        self.manager = manager
+
+    def render_GET(self, request: 'Request') -> bytes:
+        """ Get request /block_at_height/ that returns a block at height in parameter
+
+            'height': int, the height of block to get
+
+            :rtype: string (json)
+        """
+        request.setHeader(b'content-type', b'application/json; charset=utf-8')
+        set_cors(request, 'GET')
+
+        # Height parameter is required
+        parsed = parse_get_arguments(request.args, ['height'])
+        if not parsed['success']:
+            return get_missing_params_msg(parsed['missing'])
+
+        args = parsed['args']
+
+        # Height parameter must be an integer
+        try:
+            height = int(args['height'])
+        except ValueError:
+            return json.dumps({
+                'success': False,
+                'message': 'Invalid \'height\' parameter, expected an integer'
+            }).encode('utf-8')
+
+        # Get hash of the block with the height
+        block_hash = self.manager.tx_storage.get_from_block_height_index(height)
+
+        # If there is no block in the index with this height, block_hash will be None
+        if block_hash is None:
+            return json.dumps({
+                'success': False,
+                'message': 'No block with height {}.'.format(height)
+            }).encode('utf-8')
+
+        block = self.manager.tx_storage.get_transaction(block_hash)
+
+        data = {'success': True, 'block': block.to_json_extended()}
+        return json.dumps(data, indent=4).encode('utf-8')
+
+
+BlockAtHeightResource.openapi = {
+    '/block_at_height': {
+        'x-visibility': 'public',
+        'x-rate-limit': {
+            'global': [
+                {
+                    'rate': '50r/s',
+                    'burst': 100,
+                    'delay': 50
+                }
+            ],
+            'per-ip': [
+                {
+                    'rate': '3r/s',
+                    'burst': 10,
+                    'delay': 3
+                }
+            ]
+        },
+        'get': {
+            'tags': ['block'],
+            'operationId': 'block',
+            'summary': 'Get block at height',
+            'description': 'Returns the data of a block at specific height.',
```

Comment:
I'd say "in the best blockchain".

Reply:
I just realized how "best blockchain" can sound like "Hathor is the best blockchain". I'll use "best chain" since there are much more uses of "best chain" than "best blockchain" in code.